.github/workflows/release-bot.yaml

```diff
@@ -97,3 +104,40 @@ jobs:
           tag: v${{ needs.semver.outputs.version }}
           commit: ${{ github.sha }}
           prerelease: ${{ needs.look_for_release.outputs.release_type == 'prerelease' }}
+
+  create-release-branch:
+    needs:
+      - look_for_release
+      - publish-release
+      - semver
+    # NOTE: only create a release branch if the release is not a patch release
+    # or a prerelease.
+    # For patch releases, the release branch should already be in place.
+    # For prereleases, we do not want to create a release branch.
+    if: ${{ needs.look_for_release.outputs.release_found == 'true' && needs.semver.outputs.patch == '0' && needs.semver.outputs.prerelease == '' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: peterjgrainger/action-create-branch@v2.2.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          # NOTE: using the full ref name because
+          # https://github.com/peterjgrainger/action-create-branch?tab=readme-ov-file#branch
+          branch: 'refs/heads/release/v${{ needs.semver.outputs.major }}.{{ needs.semver.outputs.minor }}.x'
+          sha: '${{ github.sha }}'
+
+  create-cherry-pick-branch-to-main:
+    needs:
+      - look_for_release
+      - publish-release
+      - semver
+    if: ${{ needs.look_for_release.outputs.release_found == 'true' && needs.semver.outputs.patch != '0' && needs.semver.outputs.prerelease == '' && needs.look_for_release.outputs.release_latest == 'true' && github.ref_name != 'main' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: carloscastrojumo/github-cherry-pick-action@v1.0.1
+        with:
+          branch: main
+          title: '[cherry-pick] {old_title}'
+          body: 'Cherry picking #{old_pull_request_id} onto this branch'
```

Comment:
`.github/workflows/release-bot.yaml` is triggered on push to main or release branches. As I understand the [action docs](https://github.com/carloscastrojumo/github-cherry-pick-action?tab=readme-ov-file#action-inputs) it might happen that the special strings (`old_title` and `old_pull_request_id`) won't be populated in such a case as the action expects pull-request trigger.

Maybe we could make this a separate workflow to mitigate this? Or just accept that we won't have this relation between PRs included in the body and title?

Reply:
Good catch! Yes, it seems there's no easy way to get that relation in that place. Changed that to just include the version and commit SHA.

PTAL.